octobot_evaluators/evaluators/scripted_evaluator.py

```diff
@@ -0,0 +1,261 @@
+                                   exchange=exchange, exchange_id=exchange_id, trigger_source=trigger_source,
+                                   trigger_value=candle or kline)
+        try:
+            import octobot_trading.api as trading_api
+            if not self._are_candles_initialized:
+                self._are_candles_initialized = trading_api.are_symbol_candles_initialized(context.exchange_manager,
+                                                                                           symbol, time_frame)
+                if not self._are_candles_initialized:
+                    self.logger.debug(f"Waiting for candles to be initialized before calling script "
+                                      f"for {symbol} {time_frame}")
+                    return
+            computed_value = None
+            is_value_missing = True
+            if self.use_cache():
+                computed_value, is_value_missing = await context.get_cached_value()
+            if is_value_missing or not self._has_script_been_called_once:
+                # always call the script at least once to save plotting statements
+                await self._pre_script_call(context)
+                computed_value = await self.get_script()(context)
+                self._has_script_been_called_once = True
+            self.eval_note = computed_value
+            eval_time = None
+            if trigger_source == enums.ActivationTopics.FULL_CANDLES.value:
+                eval_time = evaluators_util.get_eval_time(full_candle=candle, time_frame=time_frame)
+            elif trigger_source == enums.ActivationTopics.IN_CONSTRUCTION_CANDLES.value:
+                eval_time = evaluators_util.get_eval_time(partial_candle=kline)
+            if eval_time is None:
+                self.logger.error("Can't compute evaluation time, using exchange time")
+                eval_time = trading_api.get_exchange_current_time(context.exchange_manager)
+            await self.evaluation_completed(cryptocurrency, symbol, time_frame,
+                                            eval_time=eval_time, context=context)
+        except ImportError:
+            self.logger.exception(f"Error when importing octobot-trading")
+        except Exception as e:
+            self.logger.exception(f"Error when calling evaluation script: {e}", True, e)
+
+    async def _pre_script_call(self, context):
+        try:
+            import octobot_trading.modes.scripting_library as scripting_library
+            # Always register activation_topics use input to enable changing it from run metadata
+            # (where user inputs are registered)
+            activation_topic_values = [
+                enums.ActivationTopics.FULL_CANDLES.value,
+                enums.ActivationTopics.IN_CONSTRUCTION_CANDLES.value
+            ]
+            await scripting_library.user_input(context, constants.CONFIG_ACTIVATION_TOPICS, "multiple-options",
```

Comment:
I think we'll have to find a better way to handle user inputs like using Services for example.

Reply:
TODO: decide where to store `scripting_library.user_input`